pkg/log/zap.go

```diff
@@ -32,3 +35,13 @@ func NewDefaultZapLogger() (logr.Logger, error) {
 	logger := zapr.NewLogger(zapLogger)
 	return logger, nil
 }
+func NewZapLoggerWithWriter(out io.Writer) logr.Logger {
+	bWriter := out
+	config := zap.NewDevelopmentConfig()
+	config.EncoderConfig.EncodeLevel = zapcore.CapitalLevelEncoder
+	config.EncoderConfig.FunctionKey = "function"
```

Comment:
on more question, why do we need these configurations, do these changes required? 

rest LGTM!!